### PR TITLE
Created Json serializable utility function

### PR DESCRIPTION
**Pull request by felixzhu17**
Json serialization loops through lists and ditionaries cleans up numpys, datetimes, pandas, and more!

Also moved utils folder to more logical place so it can be accessed everywhere

### CODE REVIEW

**Review by jtwinrelevanceai — COMMENTED** [review]
Looks good, test_json_serialize.py should be more robust, however.

File: tests/unit/test_json_serialize.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import pytest
+import random
+
+
+def test_datetime_upload(test_datetime_dataset):
+    test = test_datetime_dataset
+    assert True
```
[the remark this answers is not in the document]
Comment (felixzhu17) [reply tests/unit/test_json_serialize.py:7]: It runs the test_datetime_dataset in the conftest.py - which uploads a datetime field into the client. Do you have any suggestions on how to make it more robust?

[the remark this answers is not in the document]
Comment (jtwinrelevanceai) [reply tests/unit/test_json_serialize.py:7]: Maybe include sample cases, with nested dictionaries/lists, with different datatypes as well

**Comment by boba-and-beer** [comment]
Remove Pydantic in this PR. Just paste this in: 
```



ENCODERS_BY_TYPE: Dict[Type[Any], Callable[[Any], Any]] = {
    bytes: lambda o: o.decode(),
    Color: str,
    datetime.date: isoformat,
    datetime.datetime: isoformat,
    datetime.time: isoformat,
    datetime.timedelta: lambda td: td.total_seconds(),
    Decimal: decimal_encoder,
    Enum: lambda o: o.value,
    frozenset: list,
    deque: list,
    GeneratorType: list,
    IPv4Address: str,
    IPv4Interface: str,
    IPv4Network: str,
    IPv6Address: str,
    IPv6Interface: str,
    IPv6Network: str,
    NameEmail: str,
    Path: str,
    Pattern: lambda o: o.pattern,
    SecretBytes: str,
    SecretStr: str,
    set: list,
    UUID: str,
}
```

File: setup.py [hunk 2]
```diff
@@ -27,6 +27,7 @@ def get_version(rel_path):
     "requests>=2.0.0",
     "numpy>=1.19.0",
     "joblib>=1.0.0",
+    "pydantic",
```
Comment (boba-and-beer) [inline setup.py:30]: Needs to be removed

File: tests/conftest.py [hunk 3]
```diff
@@ -87,6 +89,85 @@ def _sample_vector_doc(doc_id: str):
...
+    return [_sample_datetime_doc(doc_id=uuid.uuid4().__str__()) for _ in range(N)]
+
+
+@pytest.fixture(scope="session")
+def sample_numpy_docs():
+    def _sample_numpy_doc(doc_id: str):
+        return {
+            "_id": doc_id,
+            "sample_1_numpy": np.random.randint(5, size=1)[0],
+            "sample_2_numpy": np.random.rand(3, 2),
```
Comment (boba-and-beer) [inline tests/conftest.py:111]: can you add `"numpy_nan": np.nan` here as well now to make sure what you added works?

**Pull request merged**